tests/test_validation_framework/test_upscaling.py

```diff
@@ -0,0 +1,120 @@
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""Test for the data averaging class"""
+
+import warnings
+
+import pandas as pd
+import numpy as np
+from datetime import datetime
+
+from pytesmo.validation_framework.upscaling import Upscaling
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore")
+
+
+def return_averager():
+    manager_parms = {
+        "datasets": None,
+        "period": None,
+        "read_ts_names": None,
+    }
+    averager = Upscaling(
+        ref_class=None,
+        others_class={},
+        upscaling_lut={},
+        manager_parms=manager_parms
+    )
+
+    return averager
+
+
+def test_upscale():
+    """Test all upscaling functions"""
+    averager = return_averager()
+
+    to_upscale = pd.concat(
+        [pd.Series(2, index=np.linspace(1,10), name='sm'),
+         pd.Series(4, index=np.linspace(1,10), name='sm')],
+        axis=1
+    )
+    # simple check of series averaging
+    upscaled = averager.upscale(to_upscale, method="average")
+    should = pd.Series(float(3), index=np.linspace(1,10))
+    assert upscaled.equals(should)
+
+
+def test_tstability():
+    """Test temporal stability filtering with noisy or uncorrelated series"""
+    averager = return_averager()
+    n_obs = 1000
+    points = np.linspace(0, 2*np.pi, n_obs)
+    ts = np.sin(points)
+    low_corr = np.sin(points+np.pi)
+    high_sterr = np.sin(points) + np.random.normal(0, 2, n_obs)
+    to_filter = pd.concat(
+        [pd.Series(ts, name='sm_1'),
+         pd.Series(ts, name='sm_2'),
+         pd.Series(ts, name='sm_3'),
+         pd.Series(low_corr, name='low_corr'),
+         pd.Series(high_sterr, name='high_sterr')],
+        axis=1
+    )
+
+    filtered = averager.tstability_filter(to_filter)
+    should = to_filter.drop(["low_corr", "high_sterr"], axis="columns")
+    assert filtered.equals(should)
+
+
+def test_temporal_matching():
+    """Test temporal matching"""
+    averager = return_averager()
+    data_ref = np.arange(30.)
+    data2match = data_ref[:-1]
+    data2match[2] = np.nan
+
+    ref_ser = pd.Series(
+        data_ref,
+        index=pd.date_range(datetime(2007, 1, 1, 0), datetime(2007, 1, 30, 0), freq="D"),
+    ).to_frame()
+    match_ser = pd.Series(
+        data2match,
+        index=pd.date_range(datetime(2007, 1, 1, 5), datetime(2007, 1, 29, 5), freq="D"),
+    ).to_frame()
+    to_match = [ref_ser, match_ser]
+
+    matched = averager.temporal_match(to_match, drop_missing=False)
+    assert len(matched.index) == 30, "Should be matched to the longest timeseries"
+
+    matched = averager.temporal_match(to_match, drop_missing=True)
+    assert len(matched.index) == 28, "Should drop the row and the missing timestep with a missing value"
+
+    matched = averager.temporal_match(to_match, hours=4)
+    assert matched[matched.columns[1]].dropna().empty, "Should not be matched"
```

Comment:
Can you also add a test for `get_upscaled_ts` that triggers the warning that you are trying to catch?

Reply:
I have decided that instead of returning None it is better to just drop the series anyway.